enhancements/agent-installer/interactive-network-config.md

```diff
@@ -0,0 +1,249 @@
+  workflow.
+
+### Non-Goals
+
+* Interactive (re)configuration of any other aspect of the installation.
+* Add interactive network configuration to CoreOS in general.
+* Allow interactive network configuration during IPI baremetal installs.
+* Create a new implementation of the network configuration available in
+  `nmtui`.
+
+## Proposal
+
+### Workflow Description
+
+The user will generate the agent ISO using `openshift-install` as usual. The
+`networkConfig` section in the `agent-config` (or the NMStateConfig ZTP
+manifests) may _or may not_ contain network configuration.
+
+The user will then boot the ISO. Any network configuration for each current
+host contained in the ISO will be applied at startup.
+
+Prior to the login prompt, a TUI will appear on the hardware console. It will
+display the current status of connectivity to the registry, which will be
+continuously updated. It will also give the user the option to interactively
+reconfigure the network, or accept the current configuration and exit.
+
+If there is connectivity available to the registry, the TUI will automatically
+exit within 20s if there is no user interaction.
+
+Services that require pulling a container image will not start until the TUI
+exits.
+
+Once the TUI exits, the login prompt will be displayed on the console as usual.
+
+### API Extensions
+
+N/A
+
+### Implementation Details/Notes/Constraints
+
+The TUI will be implemented as a golang binary and delivered in a container
+added to the release payload. The actual network configuration will be done
+through the existing `nmtui` binary, which is already present in CoreOS.
+
+To allow it to run before the network is configured, the TUI binary will be
+extracted from the release payload by the installer process, and embedded in
+the CoreOS ISO (which is also extracted from the release payload). Since even a
+modest golang binary is likely too large for the pre-allocated Ignition config
+area, the binary will be added to the initramfs along with a script in the
+`/usr/lib/dracut/hooks/pre-pivot/` directory to copy it from the initramfs to
+the system root. To add these files to the initramfs in the ISO, they will be
+added to a new CPIO archive, and this archive will be appended to the ignition
+volume *after* the empty space at the end. This allows users to continue to use
+the `coreos-installer` commands to extract the Ignition config and replace it
+with a different one. In PXE installs, the installer will append the new CPIO
+archive to the rootfs file, immediately after the CPIO archive containing the
+Ignition config.
+
+The running network configuration is copied to disk using the `--copy-network`
+argument to `coreos-installer`, so no other changes are required to persist the
+desired configuration to the network.
+
+Access to the registry could be checked by simply telling podman to pull the
+release image. However, if this fails then ideally we would be able to identify
```

Comment:
To avoid downloading initially the image the Docker registry API could be used as well